File: src/main/java/jp/aoichaan0513/A_TosoGame_Live/Listeners/onBreak.java
package jp.aoichaan0513.A_TosoGame_Live.Listeners;

import jp.aoichaan0513.A_TosoGame_Live.API.Scoreboard.Teams;
import org.bukkit.entity.EntityType;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.hanging.HangingBreakByEntityEvent;

public class onBreak implements Listener {
    @EventHandler
    public void onHangingBreak(HangingBreakByEntityEvent e) {
        if (e.getRemover().getType() == EntityType.PLAYER) {
            Player p = (Player) e.getRemover();
            if (!Teams.hasJoinedTeam(Teams.OnlineTeam.TOSO_ADMIN, p)) {
                e.setCancelled(true);
            }
        }
    }
}

File: src/main/kotlin/jp/aoichaan0513/A_TosoGame_Live/Listeners/Minecraft/onBreak.kt
package jp.aoichaan0513.A_TosoGame_Live.Listeners.Minecraft

import jp.aoichaan0513.A_TosoGame_Live.Main
import jp.aoichaan0513.A_TosoGame_Live.Utils.isAdminTeam
import org.bukkit.block.data.type.Stairs
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockBreakEvent
import org.bukkit.event.hanging.HangingBreakByEntityEvent

class onBreak : Listener {

    @EventHandler
    fun onBreak(e: BlockBreakEvent) {
        val block = e.block

        if (block.blockData !is Stairs || Main.arrowSet.none { it.location.blockX == block.x && it.location.blockY == block.y && it.location.blockZ == block.z }) return

        val armorStand = Main.arrowSet.first { it.location.blockX == block.x && it.location.blockY == block.y && it.location.blockZ == block.z }
        armorStand.remove()
        Main.arrowSet.remove(armorStand)
    }

    @EventHandler
    fun onHangingBreak(e: HangingBreakByEntityEvent) {
        val p = e.remover as? Player ?: return

        if (!p.isAdminTeam)
            e.isCancelled = true
    }
}